builder/amazon/common/ssm_driver.go

```diff
@@ -132,6 +191,7 @@ func (d *SSMDriver) openTunnelForSession(ctx context.Context) error {
 
 			if stdoutCh == nil && stderrCh == nil {
 				log.Printf("[DEBUG] %s: %s", prefix, "active session has been terminated; stopping all log polling processes.")
+				d.retryAfterTermination <- true
```

Comment:
In my testing so far I find that we are always failing into this case because the command seems to be terminating when it can't make a connection but the session is still active. We may need to adjust the log handling here to know if we should send to retryAfterTermination or retryConnection.